AP-protocol.md

```diff
@@ -98,26 +92,35 @@ To do so, they must use the **Network Discovery Protocol**. The Network Discover
 The client or server that wants to learn the topology, called the **initiator**, starts by flooding a query to all its immediate neighbors:
 
 ```rust
+enum NodeType{Client, Drone, Server}
+
 struct Query {
 	/// Unique identifier of the flood, to prevent loops.
 	flood_id: u64,
 	/// ID of client or server
-	initiator_id: RC<Arc<usize>>,
+	initiator_id: u64,
 	/// Time To Live, decremented at each hop to limit the query's lifespan.
+	/// When ttl reaches 0, we start a QueryResult message that reaches back to the initiator
 	ttl: u64,
 	/// Records the nodes that have been traversed (to track the connections).
-	path_trace: [u64; 20]
-	node_types: [NodeType; ]
+	path_trace: Vec<(u64, NodeType)>
 }
 ```
 
 ### **Neighbor Response**
 
 When a neighbor node receives the query, it processes it based on the following rules:
 
-- If the query was received earlier (i.e., the query identifier is already known), the node broadcasts the query to increase the robustness.
-- Otherwise, the node appends its own ID and type (that is, Drone, Client, Server) to the **path trace** and forwards the query to its neighbors (except the one it received the query from).
-- Optionally, the node can return an **acknowledgment** or **path information** directly to the initiator, revealing its connection in the process.
+- If the query was not received earlier, the node forwards the updated message to its neighbours (except the one it received the query from) decreasing the TTL by 1, otherwise set the TTL to 0.
```

Comment:
We could then take advantage of **received earlier** as a condition for which its possible to build a `QueryResult` (instead of waiting every time for the ttl to become 0) as it would mean the protocol has discovered a cycle in the network and it's not possible to gather further information about the topology.